content/200-concepts/100-components/01-prisma-schema/06-relations/425-referential-actions/100-cyclic-referential-actions.mdx

```diff
@@ -146,3 +146,72 @@ model Fox {
   foodStore Egg[]
 }
 ```
+
+## Multiple cascade paths between two models
+
+The data model describes two different paths between same models, with both relations triggering cascading referential actions.
+
+```prisma
+model User {
+  id        Int       @id @default(autoincrement())
+  comments  Comment[]
+  posts     Post[]
+}
+
+model Post {
+  id        Int       @id @default(autoincrement())
+  authorId  Int
+  author    User      @relation(fields: [authorId], references: [id])
+  comments  Comment[]
+}
+
+model Comment {
+  id          Int      @id @default(autoincrement())
+  writtenById Int
+  postId      Int
+  writtenBy   User     @relation(fields: [writtenById], references: [id])
+  post        Post     @relation(fields: [postId], references: [id])
+}
+```
+
+The problem in this data model is how there are two paths from `Comment` to the `User`, and how the default `onUpdate` action in both relations is `Cascade`. This leads into two validation errors:
+
+The first one is in the relation `writtenBy`:
+
+```terminal
+Error parsing attribute "@relation": When any of the records in model `User` is updated or deleted, the referential actions on the relations cascade to model `Comment` through multiple paths. Please break one of these paths by setting the `onUpdate` and `onDelete` to `NoAction`. (Implicit default `onUpdate`: `Cascade`)
```

Comment:
If we run validation, does this not also highlight the relation the error is about? Then we could add the full output here, making it even more explicit than "just" the error msg.

Reply:
We just don't get the nice colors, that's why we didn't have them here to begin with.

Reply:
Nice colors is not what I suggest, but the "context" of the model that is usually highlighted. Or does that not happen here?